src/WebJobs.Script/Binding/FunctionBinding.cs

```diff
@@ -214,6 +214,10 @@ internal static ICollection ReadAsCollection(object value)
             {
                 // Read the value into the context Value converting based on data type
                 object converted = null;
+                if(context.DataType == DataType.Stream)
```

Comment:
Is this check necessary? Having the `DataType` indicating that this is a stream doesn't guarantee the value type. The check is also performed in `ConvertStreamToValue`.
Can we bypass this if block altogether and just have:
`object converted = context.Value;` (without casting to Stream) ? 

If there's a possibility `Value` won't be set we can also allow for those scenarios by assigning a new stream.

Thoughts?